### PR TITLE
NAS-132267 / 25.04 / Convert iscsi.extent.* to new api

### PR DESCRIPTION
**Pull request by bmeagherix**
Converts the `iscsi.extent` endpoints to the new api paradigm.

----
Clear CI run [here](http://jenkins.eng.ixsystems.net:8080/job/tests/job/api_tests/1714/):

### CODE REVIEW

File: src/middlewared/middlewared/plugins/iscsi_/extents.py [hunk 1]
```diff
@@ -68,26 +68,7 @@ async def sharing_task_determine_locked(self, data, locked_datasets):
...
-        Str('comment'),
-        Bool('insecure_tpc', default=True),
-        Bool('xen'),
-        Str('rpm', enum=['UNKNOWN', 'SSD', '5400', '7200', '10000', '15000'],
-            default='SSD'),
-        Bool('ro', default=False),
-        Bool('enabled', default=True),
-        register=True
-    ), audit='Create iSCSI extent', audit_extended=lambda data: data["name"])
+    @api_method(IscsiExtentCreateArgs, IscsiExtentCreateResult, audit='Create iSCSI extent', audit_extended=lambda data: data['name'])
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/iscsi_/extents.py:71]: flake8 here (line too long)

Comment (yocalebo) [inline src/middlewared/middlewared/plugins/iscsi_/extents.py:71]:
```suggestion
    @api_method(
        IscsiExtentCreateArgs,
        IscsiExtentCreateResult,
        audit='Create iSCSI extent',
        audit_extended=lambda data: data['name']
    )
```

Comment (bmeagherix) [reply src/middlewared/middlewared/plugins/iscsi_/extents.py]: Addressed.

But FMI what is the max length permitted by your flake8?

**Comment by creatorcary** [comment]
Is there danger in adding types to api/base if those types are subject to change in the future? It is possible to import from other modules in api/v25_04_0.

File: src/middlewared/middlewared/plugins/iscsi_/extents.py (hunk 1 above)
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/iscsi_/extents.py]: 120

**Pull request merged**